src/Apps/Artwork/Components/__tests__/ArtistInfo.test.tsx

```diff
@@ -39,46 +40,64 @@ describe("ArtistInfo", () => {
+    })
   })
 
-  it("Hides 'Show artist insights' button if no market insights or selected exhibitions data", async () => {
-    const data = cloneDeep(ArtistInfoFixture)
-    data.highlights.partners = null
-    data.collections = null
-    data.auctionResults = null
-    data.exhibition_highlights = null
-    wrapper = await getWrapper(data)
-    expect(wrapper.find("Button").length).toBe(0)
-  })
+  describe("ArtistInfo for artwork with incomplete artist info", () => {
+    it("Hides 'Show artist insights' button if no market insights or selected exhibitions data", async () => {
+      const data = cloneDeep(ArtistInfoFixture)
+      data.highlights.partners = null
+      data.collections = null
+      data.auctionResults = null
+      data.exhibition_highlights = null
+      wrapper = await getWrapper(data)
+      expect(wrapper.find("Button").length).toBe(0)
+    })
 
-  it("hides ArtistBio if no data", async () => {
-    const data = cloneDeep(ArtistInfoFixture)
-    data.biography_blurb.text = null
-    wrapper = await getWrapper(data)
-    expect(wrapper.find("ArtistBio").length).toBe(0)
-  })
+    it("hides ArtistBio if no data", async () => {
+      const data = cloneDeep(ArtistInfoFixture)
+      data.biography_blurb.text = null
+      wrapper = await getWrapper(data)
+      expect(wrapper.find("ArtistBio").length).toBe(0)
+    })
 
-  it("hides MarketInsights if no data", async () => {
-    const data = cloneDeep(ArtistInfoFixture)
-    data.highlights.partners = null
-    data.collections = null
-    data.auctionResults = null
-    wrapper = await getWrapper(data)
-    wrapper.find("Button").simulate("click")
-    expect(wrapper.find("MarketInsights").html()).toBe(null)
+    it("hides MarketInsights if no data", async () => {
+      const data = cloneDeep(ArtistInfoFixture)
+      data.highlights.partners = null
+      data.collections = null
+      data.auctionResults = null
+      wrapper = await getWrapper(data)
+      wrapper.find("Button").simulate("click")
+      expect(wrapper.find("MarketInsights").html()).toBe(null)
+    })
+
+    it("hides SelectedExhibitions if no data", async () => {
+      const data = cloneDeep(ArtistInfoFixture)
+      data.exhibition_highlights = []
+      wrapper = await getWrapper(data)
+      wrapper.find("Button").simulate("click")
+      expect(wrapper.find("SelectedExhibitions").html()).toBe(null)
+    })
   })
 
-  it("hides SelectedExhibitions if no data", async () => {
-    const data = cloneDeep(ArtistInfoFixture)
-    data.exhibition_highlights = []
-    wrapper = await getWrapper(data)
-    wrapper.find("Button").simulate("click")
-    expect(wrapper.find("SelectedExhibitions").html()).toBe(null)
+  describe("Analytics", () => {
+    beforeAll(async () => {
+      wrapper = await getWrapper()
+    })
+    it("tracks click on 'Show artist insights' button", () => {
+      const button = wrapper.find("Button")
+      button.simulate("click")
+      expect(track).toBeCalledWith({
```

Comment:
Ah, there might be a gotcha here 😅 This track api is super confusing.

`track` itself just returns a decorator function which gets applied when the file `ArtisInfo.tsx` is loaded. So this assertion is only checking that the most recent attempt to annotate a method with `track` was the one for the 'show artist insights' button. It's not checking that a tracking event was dispatched.

To do that, we have this special utility called `mockTracking`. Here's an example usage:

https://github.com/artsy/reaction/blob/58c4a5d5d053dc5ca74f42359cf1668949a3307d/src/Components/Publishing/Byline/__tests__/DateSource.test.tsx#L48-L58

Note that you need to put `jest.unmock('react-tracking')` at the top of the test file for this to work.